pg/src/main/java/org/bouncycastle/openpgp/PGPSignature.java

```diff
@@ -455,7 +455,7 @@ else if (getKeyAlgorithm() == PublicKeyAlgorithmTags.EDDSA_LEGACY)
             {
                 byte[] a = BigIntegers.asUnsignedByteArray(sigValues[0].getValue());
                 byte[] b = BigIntegers.asUnsignedByteArray(sigValues[1].getValue());
-                if (a.length + b.length == Ed25519.SIGNATURE_SIZE)
```

Comment:
Here, `a.length` or `b.length` could be `31` instead of the the expected `32`.
Since Ed25519 signatures are shorter than Ed448 signatures, this comparison should hold.